cpp/src/merge/merge.cu

```diff
@@ -99,8 +101,8 @@ __global__ void materialize_merged_bitmask_kernel(
     // Only one thread writes output
     if (0 == threadIdx.x % warpSize) { out_validity[word_index(destination_row)] = result_mask; }
 
-    destination_row += blockDim.x * gridDim.x;
-    active_threads = __ballot_sync(active_threads, destination_row < num_destination_rows);
+    tid += cudf::thread_index_type{blockDim.x} * cudf::thread_index_type{gridDim.x};
```

Comment:
Maybe compute this once outside the loop and store as `stride`?

Reply:
Also if you wait for #13996 to merge you can use that function instead here. At this point we're blocked on CI by #14018 anyway.